### PR TITLE
fix file upload

### PR DESCRIPTION
**Pull request by haroun**
## Summary

Fixes file upload error

## What are the specific steps to test this change?

1. Create a form
2. Add a file upload field and a text field
3. Fill the form and submit it
4. You should not see any error messages

## What kind of change does this PR introduce?
*(Check at least one)*

- [x] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/form-file-field-widget/ui/src/index.js [hunk 1]
```diff
@@ -13,10 +13,14 @@ export default () => {
   };
 
   const sizeLimiter = (input) => {
+    if ((input.dataset.maxSize || false) === false) {
```
Comment (ETLaurent) [inline modules/@apostrophecms/form-file-field-widget/ui/src/index.js:16]:
```suggestion
    if (!input.dataset.maxSize) {
```

Right?

**Review by ETLaurent — APPROVED** [review]

File: modules/@apostrophecms/form-file-field-widget/ui/src/index.js [hunk 2]
```diff
@@ -13,10 +13,14 @@ export default () => {
...
+      return;
+    }
+
     const { files } = input;
     const totalSize = Array.from(files || []).reduce((sum, { size }) => sum + size, 0);
 
-    const units = JSON.parse(input.dataset.fileSizeUnits) || {};
+    const units = JSON.parse(input.dataset.fileSizeUnits || '') || {};
```
Comment (ValJed) [inline modules/@apostrophecms/form-file-field-widget/ui/src/index.js:23]: `JSON.parse()` of empty string triggers error in browser.
Maybe:
```javascript
    const units = JSON.parse(input.dataset.fileSizeUnits || '{}');
```

**Review by ValJed — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/form-file-field-widget/ui/src/index.js (hunk 2 above)
Comment (ETLaurent) [reply modules/@apostrophecms/form-file-field-widget/ui/src/index.js:23]: ohhh good point

Comment (haroun) [inline modules/@apostrophecms/form-file-field-widget/ui/src/index.js:23]:
```suggestion
    const units = JSON.parse(input.dataset.fileSizeUnits || '{}');
```

**Review by ETLaurent — APPROVED** [review]

**Review by ValJed — APPROVED** [review]

**Pull request merged**